diff --git a/src/game/level_update.c b/src/game/level_update.c
index c221a818..a83cf82f 100644
--- a/src/game/level_update.c
+++ b/src/game/level_update.c
@@ -53,32 +53,28 @@
 
 #define WARP_NODE_CREDITS_MIN 0xF8
 
-const char *credits01[] = { "1GAME DIRECTOR", "SHIGERU MIYAMOTO" };
-const char *credits02[] = { "2ASSISTANT DIRECTORS", "YOSHIAKI KOIZUMI", "TAKASHI TEZUKA" };
-const char *credits03[] = { "2SYSTEM PROGRAMMERS", "YASUNARI NISHIDA", "YOSHINORI TANIMOTO" };
-const char *credits04[] = { "3PROGRAMMERS", "HAJIME YAJIMA", "DAIKI IWAMOTO", "TOSHIO IWAWAKI" };
-const char *credits05[] = {
-    "4CAMERA PROGRAMMER", "MARIO FACE PROGRAMMER", "TAKUMI KAWAGOE", "GILES GODDARD"
-}; // US combines camera programmer and Mario face programmer
-const char *credits06[] = { "2COURSE DIRECTORS", "YOICHI YAMADA", "YASUHISA YAMAMURA" };
-const char *credits07[] = { "2COURSE DESIGNERS", "KENTA USUI", "NAOKI MORI" };
-const char *credits08[] = { "3COURSE DESIGNERS", "YOSHIKI HARUHANA", "MAKOTO MIYANAGA",
-                            "KATSUHIKO KANNO" };
-const char *credits09[] = { "1SOUND COMPOSER", "KOJI KONDO" };
-const char *credits10[] = { "4SOUND EFFECTS", "SOUND PROGRAMMER", "YOJI INAGAKI",
-                            "HIDEAKI SHIMIZU" }; // as well as sound effects and sound programmer
-const char *credits11[] = { "23-D ANIMATORS", "YOSHIAKI KOIZUMI", "SATORU TAKIZAWA" };
-const char *credits12[] = { "1ADDITIONAL GRAPHICS", "MASANAO ARIMOTO" };
-const char *credits13[] = { "3TECHNICAL SUPPORT", "TAKAO SAWANO", "HIROHITO YOSHIMOTO", "HIROTO YADA" };
-const char *credits14[] = { "1TECHNICAL SUPPORT", "SGI N64 PROJECT STAFF" };
-const char *credits15[] = { "2PROGRESS MANAGEMENT", "KIMIYOSHI FUKUI", "KEIZO KATO" };
-const char *credits16[] = { "5SCREEN TEXT WRITER", "TRANSLATION", "LESLIE SWAN", "MINA AKINO",
-                            "HIRO YAMADA" }; // ...in order to make room for these 2 new lines
-const char *credits17[] = { "4MARIO VOICE", "PEACH VOICE", "CHARLES MARTINET", "LESLIE SWAN" };
-const char *credits18[] = { "3SPECIAL THANKS TO", "EAD STAFF", "ALL NINTENDO PERSONNEL",
-                           "MARIO CLUB STAFF" };
-const char *credits19[] = { "1PRODUCER", "SHIGERU MIYAMOTO" };
-const char *credits20[] = { "1EXECUTIVE PRODUCER", "HIROSHI YAMAUCHI" };
+const char *credits01[] = { "1CREATIVE DIRECTOR", "ROKONSHIMO" };
+const char *credits02[] = { "3IMAGE GENERATOR", "MAX WOOLF", "KATHERINE CROWSON", "AHSEN KHALIQ" };
+const char *credits03[] = { "2IMAGE EDITOR USED", "PAINT DOT NET", "RICK BREWSTER" };
+const char *credits04[] = { "1TEXT GENERATOR", "MAX WOOLF" };
+const char *credits05[] = { "2FONTS USED", "DEEP SCRIPT 2", "MAXIM LEYZEROVICH" };
+const char *credits06[] = { "3FONTS USED", "LUCIDA CALLIGRAPHY", "CHARLES BIGELOW", "CHRIS HOLMES" };
+const char *credits07[] = { "3FONTS USED", "LINJA PONA", "DAVID ROBERTS", "JAN WESI" };
+const char *credits08[] = { "1PROGRAMMING", "RENDER NINETY-SIX TEAM" };
+const char *credits09[] = { "3COMPILING TOOL", "GUNVALK", "FILIPIANOSOL",
+                            "GAMMATENDONNINE" };
+const char *credits10[] = { "1ORIGINAL GAME", "NINTENDO" };
+const char *credits11[] = { "2WAIT ITS STILL GOING", "THERES NO MORE CREDITS", "HAVE SOME AI ONES I GUESS" };
+const char *credits12[] = { "1YOJI INEMIC UNIVERSE PLANET DESIGNER", "MARIO JULIEN MICHELS" };
+const char *credits13[] = { "23-D MARKER MARKER MARKER", "YOJIMOTO TYPE-A ADMIN", "HAZHAZARD MATTEO SEWELLS" };
+const char *credits14[] = { "1STARCARNER", "ZOMBIE JAI HAWKEYE" };
+const char *credits15[] = { "2TEXAS STREET EXECUTIVE PROGRAMMERS", "DAVE HAZELER", "GREG HAZELER" };
+const char *credits16[] = { "1JAPANESE TANK TROOPER CHARACTER CREATION", "JOHN" };
+const char *credits17[] = { "2FOOD SPOTNER", "TATSUO SATO", "SATOSHI SAITO" };
+const char *credits18[] = { "3VICMARK SOCIONICS CREWMEN", "YACHTY ISAMAYOR", "KATTUSUHIMOHO",
+                            "SUPER APART NL LAB" };
+const char *credits19[] = { "1DAZIEL ATHLETE DESIGNER", "ED GU STEROIS" };
+const char *credits20[] = { "1MARIO CLUB AGENT MAY ALSO BE MAY GOD", "JULIANNE KIMMAREJI" };
 
 struct CreditsEntry sCreditsSequence[] = {
     { LEVEL_CASTLE_GROUNDS, 1, 1, -128, { 0, 8000, 0 }, NULL },